### PR TITLE
Added GUI for Game of Life

### PR DESCRIPTION
**Pull request by annafrei924**
GameOfLife

### CODE REVIEW

File: src/main/java/freiman/conway/GridComponent.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package freiman.conway;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+
+public class GridComponent extends JComponent {
+    private Grid grid;
+    private final int gridSpacing = 15;
```
Comment (andrewoid) [inline src/main/java/freiman/conway/GridComponent.java:10]: Static final

File: src/main/java/freiman/conway/GridComponent.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        //draw the boxes
+        for (int y = 0; y < grid.getHeight(); y++) {
+            for (int x = 0; x < grid.getWidth(); x++) {
+                if (grid.isAlive(x, y)) {
+                    g.setColor(Color.GREEN);
+                    g.fillRect(x * gridSpacing, y * gridSpacing, gridSpacing, gridSpacing);
+                } else {
+                    g.setColor(Color.LIGHT_GRAY);
+                    g.drawRect(x * gridSpacing, y * gridSpacing, gridSpacing, gridSpacing);
+                }
```
Comment (andrewoid) [inline src/main/java/freiman/conway/GridComponent.java:58]: Set the whole background to light grey, therefore you just need to draw green squares.

**Review by andrewoid — COMMENTED** [review]
Add a readme and a screenshot

File: src/main/java/freiman/conway/GridComponent.java (hunk 2 above)
Comment (annafrei924) [reply src/main/java/freiman/conway/GridComponent.java:63]: Screenshot 2024-09-14 at 11 46 15 PM
Like this?

Comment (andrewoid) [reply src/main/java/freiman/conway/GridComponent.java:63]: Yes

File: src/main/java/freiman/conway/GridComponent.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public GridComponent(Grid grid) {
+        this.grid = grid;
+        addMouseListener(new MouseListener() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                if (!grid.isAlive(e.getX() / gridSpacing, e.getY() / gridSpacing)) {
+                    grid.put(e.getX() / gridSpacing, e.getY() / gridSpacing);
+                } else {
+                    grid.remove(e.getX() / gridSpacing, e.getY() / gridSpacing);
+                }
```
Comment (andrewoid) [inline src/main/java/freiman/conway/GridComponent.java:22]: call getX and getY and do the division only once and save the value. Makes this more readable.

File: src/main/java/freiman/conway/GridFrame.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package freiman.conway;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class GridFrame extends JFrame {
+    int gridSpacing = 15;
```
Comment (andrewoid) [inline src/main/java/freiman/conway/GridFrame.java:9]: private static final

**Review by andrewoid — APPROVED** [review]

**Pull request merged**